src/core_modules/capture-core/components/DataEntries/Enrollment/EnrollmentDataEntry.component.js

```diff
@@ -251,27 +289,49 @@ const getGeometrySettings = () => ({
 type FinalTeiDataEntryProps = {
     enrollmentMetadata: Enrollment,
     programId: string,
+    programCategory?: ProgramCategory,
 };
 // final step before the generic dataEntry is inserted
-class FinalEnrollmentDataEntry extends React.Component<FinalTeiDataEntryProps> {
+
+const dataEntrySectionDefinitions = {
+    [sectionKeysForEnrollmentDataEntry.ENROLLMENT]: {
+        placement: placements.TOP,
+        name: i18n.t('Enrollment'),
+    },
+    [sectionKeysForEnrollmentDataEntry.CATEGORYCOMBO]: {
+        placement: placements.TOP,
+        name: 'Data entry',
+    },
+};
+
+type State = {
+    dataEntrySections: { [string]: {name: string, placement: $Values<typeof placements>}},
+}
+class FinalEnrollmentDataEntry extends React.Component<FinalTeiDataEntryProps, State> {
+    constructor(props) {
+        super(props);
+        if (props.programCategory) {
```

Comment:
NIT: Avoid modifying the `dataEntrySectionDefinitions` object and create a new object instead. Something like this.

```
const dataEntrySections = props.programCategory
    ? {
        ...dataEntrySectionDefinitions,
        [sectionKeysForEnrollmentDataEntry.CATEGORYCOMBO]: {
          placement: placements.TOP,
          name: props.programCategory.displayName,
        },
    }
    : dataEntrySectionDefinitions;

this.state = {
    dataEntrySections,
};
```